lib/OpenQA/WebAPI/Controller/Main.pm

```diff
@@ -30,6 +30,12 @@ sub _group_result {
 
     my %res;
     my $jobs = $group->jobs->search({"me.t_created" => $timecond,});
+    # arbitrary default limit
+    my $limit_jobs = $self->param('limit_jobs') // 4000;
```

Comment:
> This still doesn't limit the jobs if limit_jobs is provided to be too big

Yes, it doesn't. So it does not prevent a DoS attack, only helps in case of too many jobs. I would not do
```perl
my $limit_jobs = min($self->param('limit_jobs'), 4000);
```
as we should still allow setting higher values for testing.  Should we have yet another limit for "safety purposes", e.g. 
```perl
if ($limit_jobs > 20000) {
    return $self->render(text => 'Exceeded valid range for parameter limit_jobs', status => 400);
}
```

> And why checking the limit after the DB query? You can add something like rows => $limit_jobs to search call.

The searching for jobs is not really costly but using a huge list in the subsequent queries, e.g. in L41 **is**.
I could also write something like:

```perl
my $limit_jobs = $self->param('limit_jobs') // 4000;
my $jobs = $group->jobs->search({"me.t_created" => $timecond,}, {rows => $limit_jobs});
if ($jobs->all == $limit_jobs) {
    $self->app->log->error('too many jobs for group ' . $group->id . ', ' . $group->name . ', truncating search from ' . $jobs->all . ' to ' . $limit_jobs);
}
```